src/main/java/com/gotocompany/depot/maxcompute/client/MaxComputeClient.java

```diff
@@ -0,0 +1,155 @@
+package com.gotocompany.depot.maxcompute.client;
+
+import com.aliyun.odps.Column;
+import com.aliyun.odps.Instance;
+import com.aliyun.odps.Odps;
+import com.aliyun.odps.OdpsException;
+import com.aliyun.odps.TableSchema;
+import com.aliyun.odps.account.Account;
+import com.aliyun.odps.account.AliyunAccount;
+import com.aliyun.odps.task.SQLTask;
+import com.aliyun.odps.tunnel.TableTunnel;
+import com.aliyun.odps.tunnel.TunnelException;
+import com.gotocompany.depot.config.MaxComputeSinkConfig;
+import com.gotocompany.depot.maxcompute.client.insert.InsertManager;
+import com.gotocompany.depot.maxcompute.client.insert.InsertManagerFactory;
+import com.gotocompany.depot.maxcompute.exception.MaxComputeTableOperationException;
+import com.gotocompany.depot.maxcompute.model.RecordWrapper;
+import com.gotocompany.depot.maxcompute.schema.SchemaDifferenceUtils;
+import com.gotocompany.depot.metrics.Instrumentation;
+import com.gotocompany.depot.metrics.MaxComputeMetrics;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Slf4j
+public class MaxComputeClient {
```

Comment:
please add unit tests for this class

Reply:
Still blocked on Odps objects not having no args constructor, Mockito not able to work with this